cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -105,4 +148,204 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+    if (supportFiles != null && supportFiles.length > 0) {
+      for (File uuidFile : supportFiles) {
+        SupportBundleOperationStatus supportBundleOperationStatus =
+          bundleGenerator.getSingleBundleJson(uuidFile.getName());
+        supportBundleOperationStatusList.add(supportBundleOperationStatus);
+      }
+    }
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(supportBundleOperationStatusList));
+  }
+
+  /**
+   * Get the specific bundle status from status.json by using uuid.
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle/{uuid}")
+  public void getSupportBundleByUUID(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId supportBundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(supportBundleEntityId, StandardPermission.GET);
+    List<SupportBundleOperationStatus> supportBundleOperationStatusList = new ArrayList<>();
+    SupportBundleOperationStatus supportBundleOperationStatus = bundleGenerator.getSingleBundleJson(uuid);
+    supportBundleOperationStatusList.add(supportBundleOperationStatus);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(supportBundleOperationStatusList));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundle/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId supportBundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(supportBundleEntityId, StandardPermission.DELETE);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    File uuidFile = new File(baseDirectory, uuid);
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    if (!uuidFile.exists()) {
+      responder.sendString(HttpResponseStatus.OK, String.format("No such uuid '%s' in Support Bundle.", uuid));
+      return;
+    }
+    bundleGenerator.deleteOldFolders(uuidFile);
+    responder.sendString(HttpResponseStatus.OK, String.format("Successfully deleted bundle %s", uuid));
+  }
+
+  /**
+   * Get the specific bundle file list or certain file content
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle/{uuid}/files")
+  public void getSupportBundleFile(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid,
+                                   @DefaultValue("") @QueryParam("folder-name") String folderName,
```

Comment:
Why use query param? If it is for downloading a specific file, the file name can be just the suffix.

Reply:
This is not a good API. Seems like this API is serving two purposes, either listing a directory or getting a file content? They should be two different API.

Reply:
because no query param selected, it should return the list of file names